tests/functional/adapter/constraints/test_constraints.py

```
@@ -0,0 +1,152 @@
+
+    @pytest.fixture
+    def int_type(self):
+        return "INTEGER"
+
+    @pytest.fixture
+    def data_types(self, schema_int_type, int_type, string_type):
+        # sql_column_value, schema_data_type, error_data_type
+        return [
+            ["1", schema_int_type, int_type],
+            ["'1'", string_type, string_type],
+            ["cast('2019-01-01' as date)", "date", "DATE"],
+            ["true", "boolean", "BOOLEAN"],
+            ["cast('2013-11-03 00:00:00-07' as TIMESTAMP)", "timestamp(6)", "TIMESTAMP"],
+            ["cast('1' as DECIMAL)", "DECIMAL", "DECIMAL"],
+        ]
+
+    @pytest.fixture(scope="class")
+    def dbt_profile_data(
+            self, unique_schema, dbt_profile_target, profiles_config_update
+    ):
+        profile = {
+            "test": {
+                "outputs": {
+                    "default": {},
+                },
+                "target": "default",
+            },
+        }
+        target = dbt_profile_target
+        target["schema"] = unique_schema
+        target["root_path"] = f"{BUCKET}.{unique_schema}"
+        profile["test"]["outputs"]["default"] = target
+
+        if profiles_config_update:
+            profile.update(profiles_config_update)
+        return profile
+
+
+class TestDremioTableConstraintsColumnsEqual(
+    DremioColumnEqualSetup, BaseTableConstraintsColumnsEqual
+):
+    pass
+
+
+class TestDremioViewConstraintsColumnsEqual(
+        DremioColumnEqualSetup,
+        BaseViewConstraintsColumnsEqual):
+    pass
+
+
+class TestDremioTableConstraintsRuntimeDdlEnforcement(
+        DremioColumnEqualSetup, BaseConstraintsRuntimeDdlEnforcement):
+    @pytest.fixture(scope="class")
+    def expected_sql(self):
+        return _expected_sql_dremio
+
+
+# TODO: Revisit once at least one of the constraints starts being enforced
+@pytest.mark.skip("Dremio does not enforce any constraints so rollbacks can't be tested")
+class TestDremioTableConstraintsRollback(BaseConstraintsRollback):
```

Once Dremio starts enforcing the `not null` constraint we can stop skipping these tests, I decided to leave them here even though skipped to future-proof this as they could be easy to miss later on

---

What is our take on skipping tests vs. marking them as explicitly failing right now?

---

For functionality that Dremio does not support we have examples where we are skipping:
https://github.com/dremio/dbt-dremio/blob/16ccee374c1fbea54ca3c10dab8096bf79ba7992/tests/functional/adapter/utils/test_array_construct.py#L20-L22

Also my idea is that we're actively handling and decreasing the expected failures so we can move away from such logic in the pipeline. I'd say we should not add to the expected failures and that it would either be this or not have the test at all (since unsupported, but by removing this it's easier to miss adding it back once Dremio starts enforcing `not null`)

---

In general, I see our strategy as - remove skips as more as possible and let the tests fail. 
The skips are fine if we have at least P2 ticket with them and we know how to fix them.

---

I'll create a ticket for these and move it to backlog. Do note that this ticket will be blocked until dremio starts enforcing `not null`

---

I'm in favor of having expected fails if it's an issue with the test itself that must be fixed, in this case it's a test with something that Dremio does not currently support so I think skipping in these cases would be best. 

DX-101860 was created